Source/Protocol/Mqtt311Handler.cpp

```diff
@@ -0,0 +1,180 @@
+#include "mqttpch.h"
+#include "Mqtt311Handler.h"
+
+#include "Rules/Rules.h"
+#include "Rules/RuleEngine.h"
+#include "Rules/ConnectValidator.h"
+#include "Rules/PublishValidator.h"
+
+#include "Converters/ConnectConverter.h"
+#include "Converters/PublishAcknowledgeConverter.h"
+#include "Converters/PublishConverter.h"
+
+namespace MQTT {
+	namespace Protocol {
+		Mqtt311Handler::Mqtt311Handler() : m_ClientStates(), m_Manager(), m_SubscribeManager()
+		{
+		}
+
+		Mqtt311Handler::~Mqtt311Handler()
+		{
+			for (auto& clientState : m_ClientStates)
```

Comment:
Deleting reference and not the pointer?

Reply:
should vector be cleaned?

Reply:
done

Reply:
Vector, doesn't have to be deleted as it is created on the stack, and will automatically be deleted when destructor is called.